components/dropdown-menu/index.js

```diff
@@ -149,9 +151,33 @@ export class DropdownMenu extends Component {
 		}
 	}
 
+	calculateMenuPosition() {
+		const { toggle } = this.nodes;
+		if ( toggle ) {
+			const node = findDOMNode( toggle );
+			let n = node;
+			let scrollLeft = 0;
+			while ( n !== null && n !== node.offsetParent ) {
+				scrollLeft += n.scrollLeft;
+				n = n.parentNode;
+			}
+			const menuLeft = node.offsetLeft - scrollLeft - 4;
+			if ( this.state.menuLeft !== menuLeft ) {
+				this.setState( { menuLeft } );
+			}
+		}
+	}
+
+	componentDidMount() {
+		this.calculateMenuPosition();
+		window.addEventListener( 'resize', this.calculateMenuPosition );
+	}
+
 	componentDidUpdate( prevProps, prevState ) {
 		const { open, activeIndex } = this.state;
 
+		this.calculateMenuPosition();
```

Comment:
Noting that this will be called after setting state from `calculateMenuPosition`, so `calculateMenuPosition` will always be called twice when it updates.

Reply:
I can't think of any way to avoid that. Do you have any suggestions?

Reply:
What are the circumstances in which we want to calculate the menu position when the component updates? If it's only when going from unopened to opened, should it go into the condition below?